File: composeApp/src/commonMain/kotlin/org/hhp227/concafe/presentation/main/ranking/RankingAction.kt
package org.hhp227.concafe.presentation.main.ranking

import org.hhp227.concafe.domain.model.RankingPeriod

sealed interface RankingAction {
    data class ChangeTab(val tab: RankingUiState.TabType) : RankingAction
    data class ChangePeriod(val period: RankingPeriod) : RankingAction
    data class ChangeRegion(val region: RankingUiState.RegionFilter) : RankingAction
    data class SelectAd(val index: Int) : RankingAction
    data class ClickMaid(val id: String) : RankingAction
    data class ClickCafe(val id: String) : RankingAction
}

File: iosApp/iosApp/Presentation/Main/Ranking/RankingAction.swift
//
//  RankingAction.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/06.
//

import Foundation
import Shared

enum RankingAction {
    case changeTab(RankingUiState.TabType)
    case changePeriod(RankingPeriod)
    case changeRegion(RankingUiState.RegionFilter)
    case selectAd(Int)
    case tapMaid(String)
    case tapCafe(String)
    case loginPromptSignInTapped
    case dismissLoginPrompt
    case updateBannerHeight(CGFloat)
}
